src/course-unit/utils.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * Adapts API URL paths to the application's internal URL format based on predefined conditions.
+ *
+ * @param {Object} params - Parameters for URL adaptation.
+ * @param {string} params.url - The original API URL to transform.
+ * @param {string} params.courseId - The course ID.
+ * @param {string} params.sequenceId - The sequence ID.
+ * @returns {string} - A correctly formatted internal route for the application.
+ */
+// eslint-disable-next-line import/prefer-default-export
+export const adoptCourseSectionUrl = (
+  { url, courseId, sequenceId }: { url: string, courseId: string, sequenceId: string },
```

Comment:
[optional]: Let's put these types into a separate interface. This should make it easier to read.

Reply:
Let's keep it closer to functionality, since these types don't seem too big.